app/src/main/kotlin/io/intrepid/fablekotlin/screens/createfable/CreateFableActivity.kt

```diff
@@ -70,6 +105,117 @@ class CreateFableActivity : BaseMvpActivity<CreateFableContract.Presenter>(), Cr
         pickedColor.setColorFilter(Color.parseColor(clickedColor.hexColor))
     }
 
+    @OnClick(R.id.circle_button1, R.id.circle_button2, R.id.circle_button3, R.id.circle_button4, R.id.circle_button5, R.id.circle_button6)
+    fun onClickCircleButton() {
+        val intent = Intent(this, SelectFriendsActivity::class.java)
+        intent.putExtra(FABLE_TITLE, enteredFableTitle.text.toString())
+        intent.putExtra(COLOR_THEME, presenter.getColorTheme())
+        intent.putExtra(ICON, (pager.currentItem + 1).toString())
+        intent.putExtra(SELECTED_FRIENDS, selectedFriends)
+        startActivityForResult(intent, PASS_FABLE_INFO)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.create_fable_menu, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        val title = enteredFableTitle.text.toString()
+        val itemClicked = item.itemId
+        when (itemClicked) {
+            R.id.createFable -> {
+                presenter.onContinueClicked(title)
+                return true
+            }
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    override fun onSupportNavigateUp(): Boolean {
+        startActivity(Intent(this, HomescreenActivity::class.java))
+        return false
+    }
+
+    // Grabs stored invitees from the other screen, fills the circles with their profile pictures
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == PASS_FABLE_INFO) {
```

Comment:
This logic should probably be in the presenter, or we should create a custom view to handle the layout.